### PR TITLE
Real master

### PR DESCRIPTION
**Pull request by idontgetoutmuch**
This should revert all the mess and keep the PureMT module (https://github.com/haskell-numerics/random-fu/issues/62). @adamConnerSax I have invited you to become a member of the group so I can add you as reviewer.

### CODE REVIEW

**Comment by adamConnerSax** [comment]
Thanks! Accepted the invite.  Happy to help!

**Comment by int-e** [comment]
For reference, the reverts restore the state after 397a115a5f71fbeade29f2a1bc0af1304171d1df.

File: random-fu/random-fu.cabal [hunk 1]
```diff
@@ -91,6 +91,7 @@ Library
   
   build-depends:        math-functions,
                         monad-loops >= 0.3.0.1,
+                        random >= 1.2.0,
```
Comment (int-e) [inline random-fu/random-fu.cabal:94]: Should we have an upper bound here?

File: random-source/random-source.cabal [hunk 2]
```diff
@@ -82,7 +82,7 @@ Library
   
   build-depends:        flexible-defaults >= 0.0.0.2,
                         mersenne-random-pure64,
-                        random,
+                        random == 1.2.0,
```
Comment (int-e) [inline random-source/random-source.cabal:85]: `==` should be `>=`, or maybe we should have `random == 1.2.*` instead (see previous comment)

File: release.nix [hunk 3]
```diff
@@ -1,15 +1,31 @@
...
+          doHaddock = false;
+        });
+
+    random-fu     = super.haskell.lib.disableLibraryProfiling (hself.callPackage  ./random-fu { });
+    random-source = super.haskell.lib.disableLibraryProfiling (hself.callPackage ./random-source { random = hsuper.random_1_2_0; });
+    rvar          = super.haskell.lib.disableLibraryProfiling (hself.callPackage ./rvar { });
+    splitmix      = hsuper.splitmix_0_1;
+    MonadRandom   = hsuper.MonadRandom_0_5_2;
+    };
+  };
```
Comment (int-e) [inline release.nix]: Not sure what's going on here. Does this pin `random` to `1.2.0`? And what goes wrong with profiling?

File: random-source/random-source.cabal [hunk 4]
```diff
@@ -1,5 +1,5 @@
 name:                   random-source
-version:                0.3.0.8
+version:                0.3.0.9
```
Comment (int-e) [inline random-source/random-source.cabal]: This should be at least `0.3.0.11`, see [hackage](https://hackage.haskell.org/package/random-source)

File: random-fu/random-fu.cabal [hunk 5]
```diff
@@ -1,5 +1,5 @@
 name:                   random-fu
-version:                0.2.7.4
+version:                0.2.7.5
```
Comment (int-e) [inline random-fu/random-fu.cabal]: And this should be at least `0.2.7.7`, see [hackage](https://hackage.haskell.org/package/random-fu)

File: random-fu/random-fu.cabal (hunk 1 above)
Comment (idontgetoutmuch) [reply random-fu/random-fu.cabal:94]: > What is the overall plan here? Is this mainly for a 0.3.0.11 release, or is the plan to abandon the generalized `RandomSource` instances from #45, in particular [ff0ce7f](https://github.com/haskell-numerics/random-fu/commit/ff0ce7fd73cd48473aa095516a2a68c01ca21ba3) (which was reverted in #55, and reinstated in #56)?

My intention is to get everything back to working and yes abandon the generalized instances until we can find a way of adding them that doesn't cause conflicts. I have no idea how difficult this is.

My personal preference for improving these packages would be to see if we could get rid of randomsource and replace it by the new random package. I'd hope we could get an ever bigger performance improvement as well as simplying the code base.

I'd be interested in your thoughts (or indeed anyone's)

Comment (idontgetoutmuch) [reply random-fu/random-fu.cabal:94]: I guess so - I'll put `>=1.2 && <1.3`

File: random-fu/random-fu.cabal (hunk 5 above)
Comment (idontgetoutmuch) [reply random-fu/random-fu.cabal]: 👍

File: release.nix (hunk 3 above)
Comment (idontgetoutmuch) [reply release.nix]: I removed it

File: random-source/random-source.cabal (hunk 4 above)
Comment (idontgetoutmuch) [reply random-source/random-source.cabal]: Done

File: random-source/random-source.cabal (hunk 2 above)
Comment (idontgetoutmuch) [reply random-source/random-source.cabal]: Done

**Comment by int-e** [comment]
I have a couple more ideas here...

* maybe squash the reverts into a single commit that mentions the state it reverts to (397a115a5f71fbeade29f2a1bc0af1304171d1df)
* we can deprecate random-fu 0.2.7.6 and random-source 0.3.0.10 on Hackage to reduce potential confusion
* unrelated to this, random-source has a partial changelog in the package description, that doesn't look like the right place (on the other hand, random-fu has a changelog.md file)

**Comment by idontgetoutmuch** [comment]
> I have a couple more ideas here...
> 
>     * maybe squash the reverts into a single commit that mentions the state it reverts to ([397a115](https://github.com/haskell-numerics/random-fu/commit/397a115a5f71fbeade29f2a1bc0af1304171d1df))
> 
>     * we can deprecate random-fu 0.2.7.6 and random-source 0.3.0.10 on Hackage to reduce potential confusion
> 
>     * unrelated to this, random-source has a partial changelog in the package description, that doesn't look like the right place (on the other hand, random-fu has a changelog.md file)

Great ideas - please go ahead with them if you have time - let me know if you need any permissions

**Pull request merged**